### DISCUSSION TITLE
Identify Nx affected apps based on file changes instead of package changes?

### DISCUSSION THREAD

**Discussion by paribhasika** [nrwl/nx#19450] [General]
In my workspace(Nx 16), we have a UI component library(with storybook) being consumed by 4 apps. There are some components being consumed by only one app (such as HeroBanner). So if I go and make some change in just HeroBanner, I would want only apps/home to come up as affected app. But since nx affected picks the changes on a package basis, it's not working out for us as desired.

> apps
> >home
> > product
> > cart

>libs
>>ui-kit
>>>HeroBanner.ts
>>>ProductCard.ts
>>>Header.ts
...

Is there any way to achieve that (making sure that the storybook implementation isn't affected) ?

**Comment by moatorres** [comment]
I think you'd have to split your components into separate libraries to achieve that. In your case would be something like: 
```sh
libs/ui-kit/hero-banner/src/HeroBanners.ts
libs/ui-kit/product-card/src/ProductCard.ts
```
so on and so forth.
